libethcore/Transaction.h

```diff
@@ -136,13 +138,14 @@ class TransactionBase
 	/// Sets the nonce to the given value. Clears any signature.
 	void setNonce(u256 const& _n) { clearSignature(); m_nonce = _n; }
 
-	/// Clears the signature.
-	void clearSignature() { m_vrs = SignatureStruct(); }
+	/// @returns true if the transaction was signed
+	bool hasSignature() const { return m_vrs.is_initialized(); }
 
-	/// @returns the signature of the transaction. Encodes the sender.
-	SignatureStruct const& signature() const { return m_vrs; }
+	/// @returns true if the transaction was signed with zero signature
+	bool hasZeroSignature() const { return m_vrs && !m_vrs->s && !m_vrs->r; }
 
-	bool hasZeroSignature() const { return !m_vrs.s && !m_vrs.r; }
+	/// @returns the signature of the transaction. Encodes the sender.
```

Comment:
it does not encode the sender right?
could we just add a comment saying that this function throw if transaction is unsigned?